internal/internal_workflow_client.go

```diff
@@ -1885,8 +1895,10 @@ func (w *workflowClientInterceptor) PollWorkflowUpdate(
 		)
 		resp, err := w.client.workflowService.PollWorkflowExecutionUpdate(ctx, &pollReq)
 		cancel()
+		var de *serviceerror.DeadlineExceeded
 		if err == context.DeadlineExceeded ||
 			status.Code(err) == codes.DeadlineExceeded ||
+			errors.As(err, &de) ||
```

Comment:
When `errors.As(err, &de)` is true, is `status.Code(err) == codes.DeadlineExceeded` not also true?

Reply:
Unfortunately, based on my testing of this scenario, no. The error returned from `PollWorkflowExecutioUpdate` when a workflow update exceeds 30s is a `*serviceerror.DeadlineExceeded`, and is not handled by `status.Code(err) == codes.DeadlineExceeded`.

Reply:
I see, the condition would have to change to `serviceerror.ToStatus(err).Code() == codes.DeadlineExceeded`. So this is fine (the other comment notwithstanding).